lib/msf/core/auxiliary/report_summary.rb

```diff
@@ -0,0 +1,131 @@
+        super(info)
+
+        if framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS)
+          register_options(
+            [
+              OptBool.new('ShowSuccessfulLogins', [false, 'Outputs a table of successful logins', true]),
+            ]
+          )
+        end
+      end
+
+      def run
+        return super unless framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS) && datastore['ShowSuccessfulLogins']
+
+        @report = {}
+        @report.extend(::Rex::Ref)
+        rhost_walker = Msf::RhostsWalker.new(datastore['RHOSTS'], datastore).to_enum
+        conditional_verbose_output(rhost_walker.count)
+        super
+        print_report_summary
+      end
+
+      def create_credential_login(credential_data)
+        return super unless framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS) && datastore['ShowSuccessfulLogins']
+
+        credential = credential_data[:core].to_credential
+        @report[rhost] = { successful_logins: [] }
+        @report[rhost][:successful_logins] << credential
+        super
+      end
+
+      # I think this will work for our new modules, but I don't know if it'll work
+      # for other modules that might not use 'session_setup' convention;
+      # i.e. you might need to hijack start_session, or create_session, or some other
+      # lower level building block that you'll have access to
+      def start_session(obj, info, ds_merge, crlf = false, sock = nil, sess = nil)
+        return super unless framework.features.enabled?(Msf::FeatureManager::SHOW_SUCCESSFUL_LOGINS) && datastore['ShowSuccessfulLogins']
+
+        result = super
+        @report[rhost].merge!({ successful_sessions: [] })
+        @report[rhost][:successful_sessions] << result
+        result
+      end
+
+      private
+
+      def print_report_summary
+        report = @report
+
+        logins = report.flat_map { |_k, v| v[:successful_logins] }.compact
+        sessions = report.flat_map { |_k, v| v[:successful_sessions] }.compact
+
+        print_status("Bruteforce completed, #{logins.size} #{logins.size == 1 ? 'credential was' : 'credentials were'} successful.")
+        if datastore['CreateSession']
+          print_status("#{sessions.size} #{sessions.size == 1 ? 'session was' : 'sessions were'} opened successfully.")
+        else
+          print_status('You can open a session with these credentials and %grnCreateSession%clr set to true')
+        end
+
+        show_successful_logins(report)
+        report
+      end
+
+      # Logic to detect if the ShowSuccessLogins datastore option has been set
+      #
+      # @param [Hash] report Host mapped to successful logins and sessions
+      # @return [String] Rex::Text::Table containing successful logins
+      def show_successful_logins(report)
+        if datastore['ShowSuccessfulLogins'] == true && !report.empty?
+          successful_logins_to_table(report)
+        end
+      end
```

Comment:
```suggestion
      def print_successful_logins(report)
        if datastore['ShowSuccessfulLogins'] == true && !report.empty?
          table = successful_logins_to_table(report)
          print_line("\n" + table.to_s + "\n")
        end
      end
```

And remove the `print_line` from `successful_logins_to_table`